### PR TITLE
Feat/scaling

### PR DESCRIPTION
**Pull request by zachferland**

### CODE REVIEW

**Comment by zachferland** [comment]
This is the rebased branch, made separate in case didn't go through well, but all looks good, and everything is up to date here now. Only thing left, im going to make oeds suggested changes on other pr, then good to merge.

File: src/__tests__/didSupport.test.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ describe('test with network', () => {
   })
 
   it('can retrieve a root store', async () => {
-    const addr = await Util.didToRootStoreAddress(DID, { ipfs, orbitdb })
-    expect(addr).toEqual(ROOT_STORE_ADDR)
+    // const addr = await Util.didToRootStoreAddress(DID, { ipfs, orbitdb })
+    // expect(addr).toEqual(ROOT_STORE_ADDR)
```
[the remark this answers is not in the document]
Comment (zachferland) [reply src/__tests__/didSupport.test.js]: nope, can be uncommented now 👍

**Comment by zachferland** [comment]
Two fixes, one that is also present in current infra. 

First
Don't invalidate cache on open, mentioned other day as unnecessary, which is true, but was also with two api nodes makes it was worse, as one node would open and write to cache, then second node would get request and also open then delete cache entry. Wasn't breaking anything, but unexpected cache behavior with two nodes. 

Two
OpenDB has some concurrency issues. If a single store (most obvious with a roostore), has openDB called on it multiple times at roughly the same time by multiple api calls, then some calls would hang. In opendb, if one call was opening, and that open db promise was not resolved yet, the responseFn would never be called. Then similarly if db was not ready (on ready event), it would call responseFn but then it would get no data from db (maybe why kenzo had seen list-spaces return empty array sometimes). Now open and ready event wrapped in single promise, and responseFn always called once db ready, whether first call or called during loading, or after.

Changes so far broke tests, have to review that still.

**Comment by oed** [comment]
At first glance it looks ok, but want to take a closer look at the concurrency code!

**Comment by zachferland** [comment]
Ok tests were just the cache change, sounds good, yeah i also need to review one more time, works so far, also these changes are at node.3box.io already

**Pull request merged**